[2024-11-04T23:25:56.078+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-11-04T23:25:54.296004+00:00 [queued]>
[2024-11-04T23:25:56.087+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-11-04T23:25:54.296004+00:00 [queued]>
[2024-11-04T23:25:56.088+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-11-04T23:25:56.099+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): reddit_extraction> on 2024-11-04 23:25:54.296004+00:00
[2024-11-04T23:25:56.104+0000] {standard_task_runner.py:57} INFO - Started process 55 to run task
[2024-11-04T23:25:56.107+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'etl_reddit_pipeline', 'reddit_extraction', 'manual__2024-11-04T23:25:54.296004+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmplupyxreh']
[2024-11-04T23:25:56.111+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask reddit_extraction
[2024-11-04T23:25:56.161+0000] {task_command.py:416} INFO - Running <TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-11-04T23:25:54.296004+00:00 [running]> on host 670f27e5f990
[2024-11-04T23:25:56.518+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Bruno Xie' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='reddit_extraction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-04T23:25:54.296004+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-04T23:25:54.296004+00:00'
[2024-11-04T23:25:56.524+0000] {logging_mixin.py:151} WARNING - Version 7.7.1 of praw is outdated. Version 7.8.1 was released Friday October 25, 2024.
[2024-11-04T23:25:56.526+0000] {logging_mixin.py:151} INFO - connected to Reddit!
[2024-11-04T23:25:57.595+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipelines/reddit_pipeline.py", line 14, in reddit_pipeline
    load_data_to_csv(post_df, OUTPUT_PATH)
  File "/opt/airflow/etls/reddit_etl.py", line 40, in load_data_to_csv
    data.to_csv(path, index=False)
  File "/home/airflow/.local/lib/python3.9/site-packages/pandas/core/generic.py", line 3902, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/home/airflow/.local/lib/python3.9/site-packages/pandas/io/formats/format.py", line 1152, in to_csv
    csv_formatter.save()
  File "/home/airflow/.local/lib/python3.9/site-packages/pandas/io/formats/csvs.py", line 247, in save
    with get_handle(
  File "/home/airflow/.local/lib/python3.9/site-packages/pandas/io/common.py", line 863, in get_handle
    handle = open(
IsADirectoryError: [Errno 21] Is a directory: '/opt/airflow/data/output'
[2024-11-04T23:25:57.618+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=etl_reddit_pipeline, task_id=reddit_extraction, execution_date=20241104T232554, start_date=20241104T232556, end_date=20241104T232557
[2024-11-04T23:25:57.629+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 6 for task reddit_extraction ([Errno 21] Is a directory: '/opt/airflow/data/output'; 55)
[2024-11-04T23:25:57.655+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-11-04T23:25:57.673+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
